<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
  <class>AdminDialog</class>
  <widget class="QDialog" name="AdminDialog">
    <property name="geometry">
      <rect>
        <x>0</x>
        <y>0</y>
        <width>560</width>
        <height>420</height>
      </rect>
    </property>
    <property name="windowTitle">
      <string>Admin Settings</string>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin"><number>10</number></property>
      <property name="topMargin"><number>0</number></property>
      <property name="rightMargin"><number>10</number></property>
      <property name="bottomMargin"><number>10</number></property>

      <!-- Title bar -->
      <item>
        <widget class="QFrame" name="customTitleBar">
          <property name="frameShape"><enum>QFrame::NoFrame</enum></property>
          <layout class="QHBoxLayout" name="customTitleLayout">
            <property name="leftMargin"><number>0</number></property>
            <property name="topMargin"><number>0</number></property>
            <property name="rightMargin"><number>0</number></property>
            <property name="bottomMargin"><number>10</number></property>
            <item>
              <widget class="QLabel" name="customTitle">
                <property name="text"><string>ADMIN SETTINGS</string></property>
              </widget>
            </item>
            <item>
              <spacer name="titleSpacer">
                <property name="orientation"><enum>Qt::Horizontal</enum></property>
                <property name="sizeHint" stdset="0">
                  <size><width>40</width><height>20</height></size>
                </property>
              </spacer>
            </item>
            <item alignment="Qt::AlignRight|Qt::AlignTop">
              <widget class="QPushButton" name="customCloseBtn">
                <property name="text"><string>√ó</string></property>
                <property name="minimumSize"><size><width>28</width><height>28</height></size></property>
                <property name="maximumSize"><size><width>28</width><height>28</height></size></property>
                <property name="flat"><bool>true</bool></property>
                <property name="toolTip"><string>Close</string></property>
              </widget>
            </item>
          </layout>
        </widget>
      </item>

      <!-- Logged-in info -->
      <item>
        <widget class="QLabel" name="loggedInLabel">
          <property name="text"><string>Logged in as: Admin</string></property>
          <property name="alignment"><set>Qt::AlignLeft|Qt::AlignVCenter</set></property>
        </widget>
      </item>

      <!-- Tabs -->
      <item>
        <widget class="QTabWidget" name="tabWidget">
          <property name="tabPosition"><enum>QTabWidget::North</enum></property>
          <property name="currentIndex"><number>0</number></property>

          <!-- Admin tab -->
          <widget class="QWidget" name="tabAdmin">
            <attribute name="title"><string>ADMIN</string></attribute>
            <layout class="QVBoxLayout" name="tabAdminLayout">
              <property name="leftMargin"><number>8</number></property>
              <property name="topMargin"><number>8</number></property>
              <property name="rightMargin"><number>8</number></property>
              <property name="bottomMargin"><number>8</number></property>

              <item>
                <widget class="QGroupBox" name="groupAdminPassword">
                  <property name="title"><string>Change Admin Password</string></property>
                  <layout class="QFormLayout" name="formAdmin">
                    <property name="labelAlignment"><set>Qt::AlignRight|Qt::AlignVCenter</set></property>

                    <item row="0" column="0">
                      <widget class="QLabel" name="labelAdminCurrent">
                        <property name="text"><string>Current Password:</string></property>
                      </widget>
                    </item>
                    <item row="0" column="1">
                      <layout class="QHBoxLayout" name="layoutAdminCurrent">
                        <property name="spacing"><number>6</number></property>
                        <item>
                          <widget class="QLineEdit" name="adminCurrentPassword">
                            <property name="echoMode"><enum>QLineEdit::Password</enum></property>
                            <property name="placeholderText"><string>********</string></property>
                          </widget>
                        </item>
                        <item>
                          <widget class="QToolButton" name="btnToggleAdminCurrent">
                            <property name="text"><string>üëÅ</string></property>
                            <property name="checkable"><bool>true</bool></property>
                            <property name="autoRaise"><bool>true</bool></property>
                            <property name="toolTip"><string>Show/Hide</string></property>
                          </widget>
                        </item>
                      </layout>
                    </item>

                    <item row="1" column="0">
                      <widget class="QLabel" name="labelAdminNew">
                        <property name="text"><string>New Password:</string></property>
                      </widget>
                    </item>
                    <item row="1" column="1">
                      <widget class="QLineEdit" name="adminNewPassword">
                        <property name="echoMode"><enum>QLineEdit::Password</enum></property>
                        <property name="placeholderText"><string>Enter new password</string></property>
                      </widget>
                    </item>
                  </layout>
                </widget>
              </item>

              <item>
                <layout class="QHBoxLayout" name="layoutAdminSave">
                  <item>
                    <spacer name="adminSaveSpacer">
                      <property name="orientation"><enum>Qt::Horizontal</enum></property>
                      <property name="sizeHint" stdset="0"><size><width>40</width><height>20</height></size></property>
                    </spacer>
                  </item>
                  <item>
                    <widget class="QPushButton" name="btnAdminSave">
                      <property name="text"><string>SAVE CHANGES</string></property>
                    </widget>
                  </item>
                  <item>
                    <spacer name="adminSaveSpacer2">
                      <property name="orientation"><enum>Qt::Horizontal</enum></property>
                      <property name="sizeHint" stdset="0"><size><width>40</width><height>20</height></size></property>
                    </spacer>
                  </item>
                </layout>
              </item>
            </layout>
          </widget>

          <!-- Staff tab -->
          <widget class="QWidget" name="tabStaff">
            <attribute name="title"><string>STAFF</string></attribute>
            <layout class="QVBoxLayout" name="tabStaffLayout">
              <property name="leftMargin"><number>8</number></property>
              <property name="topMargin"><number>8</number></property>
              <property name="rightMargin"><number>8</number></property>
              <property name="bottomMargin"><number>8</number></property>

              <item>
                <widget class="QGroupBox" name="groupStaffPassword">
                  <property name="title"><string>Change Staff Password</string></property>
                  <layout class="QFormLayout" name="formStaff">
                    <property name="labelAlignment"><set>Qt::AlignRight|Qt::AlignVCenter</set></property>

                    <item row="0" column="0">
                      <widget class="QLabel" name="labelStaffCurrent">
                        <property name="text"><string>Current Password:</string></property>
                      </widget>
                    </item>
                    <item row="0" column="1">
                      <layout class="QHBoxLayout" name="layoutStaffCurrent">
                        <property name="spacing"><number>6</number></property>
                        <item>
                          <widget class="QLineEdit" name="staffCurrentPassword">
                            <property name="echoMode"><enum>QLineEdit::Password</enum></property>
                            <property name="placeholderText"><string>********</string></property>
                          </widget>
                        </item>
                        <item>
                          <widget class="QToolButton" name="btnToggleStaffCurrent">
                            <property name="text"><string>üëÅ</string></property>
                            <property name="checkable"><bool>true</bool></property>
                            <property name="autoRaise"><bool>true</bool></property>
                            <property name="toolTip"><string>Show/Hide</string></property>
                          </widget>
                        </item>
                      </layout>
                    </item>

                    <item row="1" column="0">
                      <widget class="QLabel" name="labelStaffNew">
                        <property name="text"><string>New Password:</string></property>
                      </widget>
                    </item>
                    <item row="1" column="1">
                      <widget class="QLineEdit" name="staffNewPassword">
                        <property name="echoMode"><enum>QLineEdit::Password</enum></property>
                        <property name="placeholderText"><string>Enter new password</string></property>
                      </widget>
                    </item>
                  </layout>
                </widget>
              </item>

              <item>
                <layout class="QHBoxLayout" name="layoutStaffSave">
                  <item>
                    <spacer name="staffSaveSpacer">
                      <property name="orientation"><enum>Qt::Horizontal</enum></property>
                      <property name="sizeHint" stdset="0"><size><width>40</width><height>20</height></size></property>
                    </spacer>
                  </item>
                  <item>
                    <widget class="QPushButton" name="btnStaffSave">
                      <property name="text"><string>SAVE CHANGES</string></property>
                    </widget>
                  </item>
                  <item>
                    <spacer name="staffSaveSpacer2">
                      <property name="orientation"><enum>Qt::Horizontal</enum></property>
                      <property name="sizeHint" stdset="0"><size><width>40</width><height>20</height></size></property>
                    </spacer>
                  </item>
                </layout>
              </item>
            </layout>
          </widget>

          <!-- Email tab -->
          <widget class="QWidget" name="tabEmail">
            <attribute name="title"><string>EMAIL</string></attribute>
            <layout class="QVBoxLayout" name="tabEmailLayout">
              <property name="leftMargin"><number>8</number></property>
              <property name="topMargin"><number>8</number></property>
              <property name="rightMargin"><number>8</number></property>
              <property name="bottomMargin"><number>8</number></property>

              <item>
                <widget class="QGroupBox" name="groupEmail">
                  <property name="title"><string>Email Settings</string></property>
                  <layout class="QFormLayout" name="formEmail">
                    <property name="labelAlignment"><set>Qt::AlignRight|Qt::AlignVCenter</set></property>

                    <item row="0" column="0">
                      <widget class="QLabel" name="labelCurrentEmail">
                        <property name="text"><string>Current Email:</string></property>
                      </widget>
                    </item>
                    <item row="0" column="1">
                      <widget class="QLineEdit" name="currentEmailLineEdit">
                        <property name="readOnly"><bool>true</bool></property>
                        <property name="placeholderText"><string>user@example.com</string></property>
                      </widget>
                    </item>

                    <item row="1" column="0">
                      <widget class="QLabel" name="labelNewEmail">
                        <property name="text"><string>New Email:</string></property>
                      </widget>
                    </item>
                    <item row="1" column="1">
                      <widget class="QLineEdit" name="newEmailLineEdit">
                        <property name="placeholderText"><string>Enter new recovery email</string></property>
                      </widget>
                    </item>
                  </layout>
                </widget>
              </item>

              <item>
                <layout class="QHBoxLayout" name="layoutEmailSave">
                  <item>
                    <spacer name="emailSaveSpacer">
                      <property name="orientation"><enum>Qt::Horizontal</enum></property>
                      <property name="sizeHint" stdset="0"><size><width>40</width><height>20</height></size></property>
                    </spacer>
                  </item>
                  <item>
                    <widget class="QPushButton" name="btnEmailSave">
                      <property name="text"><string>SAVE CHANGES</string></property>
                    </widget>
                  </item>
                  <item>
                    <spacer name="emailSaveSpacer2">
                      <property name="orientation"><enum>Qt::Horizontal</enum></property>
                      <property name="sizeHint" stdset="0"><size><width>40</width><height>20</height></size></property>
                    </spacer>
                  </item>
                </layout>
              </item>
            </layout>
          </widget>
        </widget>
      </item>

      <!-- Footer info and close -->
      <item>
        <widget class="QLabel" name="infoLabel">
          <property name="text"><string>Only Admin can modify settings</string></property>
          <property name="alignment"><set>Qt::AlignLeft|Qt::AlignVCenter</set></property>
        </widget>
      </item>
      <item>
        <layout class="QHBoxLayout" name="footerLayout">
          <item>
            <spacer name="footerSpacer">
              <property name="orientation"><enum>Qt::Horizontal</enum></property>
              <property name="sizeHint" stdset="0"><size><width>40</width><height>20</height></size></property>
            </spacer>
          </item>
          <item>
            <widget class="QPushButton" name="closeButton">
              <property name="text"><string>CLOSE</string></property>
            </widget>
          </item>
        </layout>
      </item>
    </layout>
  </widget>
  <resources/>
  <connections>
    <!--
      Scaffold: You can wire signals here if you prefer Designer-based connections.
      We currently wire the eye buttons in Python (modules/menu/admin_menu.py).

      Example (for reference only; not added to avoid type-mismatch):
        <connection>
          <sender>btnToggleAdminCurrent</sender>
          <signal>toggled(bool)</signal>
          <receiver>adminCurrentPassword</receiver>
          <slot>setEchoMode(QLineEdit::EchoMode)</slot>
        </connection>

      Note: setEchoMode expects an EchoMode enum, not a bool; the Python scaffold
      maps bool ‚Üí Password/Normal correctly. Keep this block as a placeholder if
      you plan to add Designer connections later.
    -->
  </connections>
</ui>
